### PR TITLE
Test helper

### PR DESCRIPTION
**Pull request by pawelkaczmarekuk**
Sample of how to user em-test-helper

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/AnnotationScenarios.java [hunk 1]
```diff
@@ -4,30 +4,45 @@
...
 
-import static org.hamcrest.CoreMatchers.*;
 import java.util.UUID;
 
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+
+@SpringBootTest(classes = {TestUtil.class, EmTestConfig.class})
+@PropertySource(value = "classpath:application.yml")
+@RunWith(SpringRunner.class)
```
Comment (linusnorton) [inline src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/AnnotationScenarios.java]: I really tried hard not to boot spring for these tests, but I guess you can fight it for ever

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/testutil/TestUtil.java [hunk 2]
```diff
@@ -2,36 +2,56 @@
...
+            .header("ServiceAuthorization", s2sHelper.getS2sToken());
+    }
+
+    private void printAllEnv() {
+        System.out.println(String.format("SYS_VARS"));
+        System.out.println(String.format("testUrl=%s", testUrl));
+        System.out.println(String.format("idamAPIUrl=%s", idamAPIUrl));
+        System.out.println(String.format("idamClientId=%s", idamClientId));
+        System.out.println(String.format("idamClientSecret=%s", idamClientSecret));
+        System.out.println(String.format("idamClientRedirectUrl=%s", idamClientRedirectUrl));
```
Comment (linusnorton) [inline src/aat/java/uk/gov/hmcts/reform/em/annotation/testutil/TestUtil.java]: Maybe use the logger?

File: src/aat/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+---
+spring:
```
Comment (linusnorton) [inline src/aat/resources/application.yml]: How did you get this to not try to connect to a database? When I did something similar it inherited the main application.yaml config and started connecting to the DB

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/AnnotationScenarios.java (hunk 1 above)
Comment (pawelkaczmarekuk) [reply src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/AnnotationScenarios.java]: Indeed. All the reform client libs (idam, dm, ccd) use spring context

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/testutil/TestUtil.java (hunk 2 above)
Comment (pawelkaczmarekuk) [reply src/aat/java/uk/gov/hmcts/reform/em/annotation/testutil/TestUtil.java]: I'll remove it - it's was just debugging.

File: src/aat/resources/application.yml (hunk 3 above)
Comment (pawelkaczmarekuk) [reply src/aat/resources/application.yml]: When you specify Bean Classes with SpringBootTest only those are loaded - the application itself is not loaded. Of course, if one of those is Configuration which creates more Beans then those are also created.

**Pull request merged**